### PR TITLE
MCGA-8 Fixed client’s issue: Small building name should only show up at a higher zoom and Implemented sqlite

### PR DESCRIPTION
**Pull request by Taimoorrana1**
1- Fixed client’s issue: Small building name should only show up at a higher zoom (street level zoom). Applied Observer pattern to notify building objects of the map’s current zoom level.
2- Buildings are now clickable
3- Setup sqlite (DatabaseHelper class) with Singleton pattern
4- Transferred Building data that was in the Campus class to database (building table).

### CODE REVIEW

File: app/src/main/java/com/concordia/mcga/fragments/NavigationFragment.java [hunk 1]
```diff
@@ -95,11 +97,25 @@ private void addBuildingMarkers() {
         for (Building building : loyBuildings) {
             createBuildingMarkersAndPolygonOverlay(building);
         }
+        map.setOnPolygonClickListener(new GoogleMap.OnPolygonClickListener() {
+            @Override
+            public void onPolygonClick(Polygon polygon) {
+                ((MainActivity) getActivity()).createToast("Building Clicked");
```
Comment (adriannadiaz) [inline app/src/main/java/com/concordia/mcga/fragments/NavigationFragment.java]: I think these may be useful for debugging purposes... but are we leaving them in Master?

Comment (Taimoorrana1) [inline app/src/main/java/com/concordia/mcga/fragments/NavigationFragment.java]: It was not for debugging purpose. I put this code there so people can know how to read a click event from a building. The toast should be replaced by the building popup UI.

File: app/src/main/java/com/concordia/mcga/fragments/NavigationFragment.java [hunk 2]
```diff
@@ -70,37 +74,49 @@ public void onMapReady(GoogleMap googleMap) {
...
+        map.addPolygon(building.getPolygonOverlayOptions()).setClickable(true);
+
+        Marker marker = map.addMarker(building.getMarkerOptions());
+        building.setMarker(marker);
     }
 
     private void populateCampuses() {
-        Campus.SGW.populateCampusWithBuildings();
-        Campus.LOYOLA.populateCampusWithBuildings();
+        Campus.SGW.populateCampusesWithBuildings();
```
Comment (Taimoorrana1) [inline app/src/main/java/com/concordia/mcga/fragments/NavigationFragment.java]: I don't populate the campus depending on which one is the "current" one. They both get populated once, when the view is created. However, thank you for pointing this line out. This line currently populate both campuses even thought it seems like only sgw should be populated. I will fix this.

File: app/src/main/java/com/concordia/mcga/helperClasses/DatabaseHelper.java [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+import java.io.OutputStream;
+import java.sql.SQLException;
+
+/**
+ * Part of the code was taken from https://blog.reigndesign.com/blog/using-your-own-sqlite-database-in-android-applications/
+ */
+
+public class DatabaseHelper extends SQLiteOpenHelper {
+    private static final String DATABASE_NAME = "development.db";
+    public static SQLiteDatabase db;
```
Comment (Taimoorrana1) [inline app/src/main/java/com/concordia/mcga/helperClasses/DatabaseHelper.java]: LOL. I'll fix it

File: app/src/main/java/com/concordia/mcga/helperClasses/DatabaseHelper.java [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+
+        //transfer bytes from the inputfile to the outputfile
+        byte[] buffer = new byte[1024];
+        int length;
+        while ((length = myInput.read(buffer)) > 0) {
+            myOutput.write(buffer, 0, length);
+        }
+
+        //Close the streams
+        myOutput.flush();
```
Comment (Taimoorrana1) [inline app/src/main/java/com/concordia/mcga/helperClasses/DatabaseHelper.java]: Good to know. Thanks. Will fix

File: app/src/main/java/com/concordia/mcga/helperClasses/DatabaseHelper.java [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+            return databaseHelper;
+        }
+
+    }
+
+    /**
+     * Creates a empty database on the system and rewrites it with your own database.
+     */
+    public void createDataBase() throws IOException {
+        boolean dbExist = checkDataBase();
```
Comment (Taimoorrana1) [inline app/src/main/java/com/concordia/mcga/helperClasses/DatabaseHelper.java]: Programming wise, I agree it is unnecessary but it makes the code easier to understand. If(!checkDataBase) doesn't make much sense to me. It's just personal choice.

**Pull request merged**